### PR TITLE
New API to load from CDN endpoint

### PR DESCRIPTION
**Pull request by zhiyuanliang-ms**

### CODE REVIEW

File: src/load.ts [hunk 1]
```diff
@@ -78,6 +78,54 @@ export async function load(
     }
 }
 
+/**
+ * Loads the data from a CDN and returns an instance of AzureAppConfiguration.
+ * @param endpoint  The URL to the CDN.
+ * @param options  Optional parameters.
+ */
+export async function loadCdn(endpoint: URL | string, options?: AzureAppConfigurationOptions): Promise<AzureAppConfiguration>;
```
Comment (zhenlan) [inline src/load.ts:86]: I'm wondering if it makes more sense to name it `loadFromCDN` insead.

Comment (Eskibear) [reply src/load.ts:86]: +1

File: src/load.ts [hunk 2]
```diff
@@ -78,6 +78,54 @@ export async function load(
...
+            endpoint = new URL(endpoint);
+        } catch (error) {
+            if (error.code === "ERR_INVALID_URL") {
+                throw new Error("Invalid endpoint URL.", { cause: error });
+            } else {
+                throw error;
+            }
+        }
+    }
+    const emptyTokenCredential: TokenCredential = {
```
Comment (avanigupta) [inline src/load.ts:104]: I think we can just call `load(endpoint, emptyTokenCredential, appConfigOptions)` without any extra validations here. `load` takes care of everything else.

File: src/load.ts (hunk 1 above)
Comment (zhiyuanliang-ms) [reply src/load.ts]: I will use camelCase, so it will be `loadFromCdn`

File: src/load.ts [hunk 3]
```diff
@@ -78,6 +78,23 @@ export async function load(
...
+ * @param options  Optional parameters.
+ */
+export async function loadFromCdn(endpoint: URL | string, options?: AzureAppConfigurationOptions): Promise<AzureAppConfiguration>;
+
+export async function loadFromCdn(
+    endpoint: string | URL,
+    appConfigOptions?: AzureAppConfigurationOptions
+): Promise<AzureAppConfiguration> {
+    const emptyTokenCredential: TokenCredential = {
+        getToken: async () => ({ token: "", expiresOnTimestamp: 0 })
```
Comment (avanigupta) [inline src/load.ts:93]: do we need to override the sync version?

Comment (zhiyuanliang-ms) [reply src/load.ts:93]: image

The `TokenCredential` interface requires getToken to be async

**Review by avanigupta — APPROVED** [review]

File: src/load.ts [hunk 4]
```diff
@@ -78,6 +78,23 @@ export async function load(
...
 
+/**
+ * Loads the data from a CDN and returns an instance of AzureAppConfiguration.
+ * @param endpoint  The URL to the CDN.
+ * @param options  Optional parameters.
+ */
+export async function loadFromCdn(endpoint: URL | string, options?: AzureAppConfigurationOptions): Promise<AzureAppConfiguration>;
+
+export async function loadFromCdn(
+    endpoint: string | URL,
```
[the remark this answers is not in the document]
Comment (zhiyuanliang-ms) [reply src/load.ts:89]: I am ok with `cdnEndpoint`. Updated.

File: src/load.ts [hunk 5]
```diff
@@ -82,6 +82,23 @@ export async function load(
     }
 }
 
+/**
+ * Loads the data from a CDN and returns an instance of AzureAppConfiguration.
+ * @param cdnEndpoint  The URL to the CDN.
+ * @param appConfigOptions  Optional parameters.
+ */
+export async function loadFromCdn(endpoint: URL | string, options?: AzureAppConfigurationOptions): Promise<AzureAppConfiguration>;
```
Comment (zhenlan) [inline src/load.ts:90]: > endpoint

Should this be updated too?

Comment (zhiyuanliang-ms) [reply src/load.ts:90]: weird, I remembered I updated this in the commit, will update it again

**Comment by zhiyuanliang-ms** [comment]
Verified Azure Frond Door + sas token can get kvs from app config. So merge this PR.

**Pull request merged**